internal/commands/edit.go

```diff
@@ -0,0 +1,325 @@
+			meeting,
+			postMortem,
+			startDate,
+			rootCause,
+			description,
+		},
+	}
+
+	return app.Client.OpenDialog(triggerID, dialog)
+}
+
+// EditIncidentByDialog starts an incident after receiving data from a Slack dialog
+func EditIncidentByDialog(
+	ctx context.Context,
+	app *app.App,
+	incidentDetails bot.DialogSubmission,
+) error {
+	app.Logger.Info(
+		ctx,
+		"command/open.EditIncidentByDialog",
+		log.NewValue("incident_edit_details", incidentDetails),
+	)
+
+	var (
+		userID        = incidentDetails.User.ID
+		channelID     = incidentDetails.Channel.ID
+		channelName   = incidentDetails.Channel.Name
+		submission    = incidentDetails.Submission
+		incidentTitle = submission["incident_title"]
+		product       = submission["product"]
+		commander     = submission["incident_commander"]
+		severityLevel = submission["severity_level"]
+		meeting       = submission["meeting_url"]
+		postMortem    = submission["post_mortem_url"]
+		initDateText  = submission["init_date"]
+		rootCause     = submission["root_cause"]
+		description   = submission["incident_description"]
+		supportTeam   = config.Env.SupportTeam
+		dateLayout    = "2006-01-02T15:04:05-0700"
+		initDate      time.Time
+	)
+
+	incidentBeforeEdit, err := app.IncidentRepository.GetIncident(ctx, channelID)
+	if err != nil {
+		return err
+	}
+
+	user, err := getSlackUserInfo(ctx, app, commander)
+	if err != nil {
+		return fmt.Errorf("commands.EditIncidentByDialog.get_slack_user_info: incident=%v commanderId=%v error=%v", channelName, commander, err)
+	}
+
+	severityLevelInt64, err := getStringInt64(severityLevel)
+	if err != nil {
+		return err
+	}
+
+	initDate, err = time.Parse(dateLayout, initDateText)
+	if err != nil {
+		app.Logger.Error(
+			ctx,
+			"command.EditIncidentByDialog Parse ERROR",
+			log.NewValue("channelID", channelID),
+			log.NewValue("initDateText", initDateText),
+			log.NewValue("error", err),
+		)
+
+		PostErrorAttachment(ctx, app, channelID, userID, err.Error())
+		return err
+	}
+
+	incident := model.Incident{
+		ID:                 incidentBeforeEdit.ID,
+		Title:              incidentTitle,
+		Product:            product,
+		DescriptionStarted: description,
+		StartTimestamp:     &initDate,
+		SeverityLevel:      severityLevelInt64,
+		CommanderID:        user.SlackID,
+		CommanderEmail:     user.Email,
+		MeetingURL:         meeting,
+		PostMortemURL:      postMortem,
+		RootCause:          rootCause,
+	}
+
+	err = app.IncidentRepository.UpdateIncident(ctx, &incident)
+
+	if err != nil {
+		return err
+	}
+
+	if incidentBeforeEdit.CommanderID != incident.CommanderID ||
+		incidentBeforeEdit.PostMortemURL != incident.PostMortemURL ||
+		incidentBeforeEdit.MeetingURL != incident.MeetingURL {
+		fillTopic(ctx, app, incident, channelID, meeting, postMortem)
+	}
+
+	attachment := createEditAttachment(incident, incidentBeforeEdit.ID, meeting, supportTeam)
+	message := "An Incident has been edited by <@" + incidentDetails.User.Name + ">"
```

Comment:
Replace "An Incident [...]" for "The incident [...]" because it will be posted in the incident channel.